File: src/main/scala/li/cil/oc/CreativeTab.scala
package li.cil.oc

import li.cil.oc.common.init.Items
import net.minecraft.item.ItemGroup
import net.minecraft.item.ItemStack
import net.minecraft.util.NonNullList

object CreativeTab extends ItemGroup(OpenComputers.Name) {
  private lazy val stack = api.Items.get(Constants.BlockName.CaseTier1).createItemStack(1)

  override def makeIcon = stack

  override def fillItemList(list: NonNullList[ItemStack]) {
    super.fillItemList(list)
    Items.decorateCreativeTab(list)
  }
}

File: src/main/scala/li/cil/oc/CreativeTab.java
package li.cil.oc;

import li.cil.oc.api.Items;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public final class CreativeTab extends CreativeTabs {
    public static final CreativeTab INSTANCE = new CreativeTab();

    private CreativeTab() {
        super(OpenComputers.ID());
    }

    @SideOnly(Side.CLIENT)
    @Override
    public ItemStack getTabIconItem() {
        return Items.get(Constants.BlockName.CaseTier1()).createItemStack(1);
    }
}
